### ISSUE TITLE
`--docker-image` should imply `--docker`

### ISSUE THREAD

**Issue by angerman** [commercialhaskell/stack#1946]
Either I'm missing something, or I'm doing something wrong. Both are likely.

I have two docker images, purely to build for different deployments. Primarily I don't want to build with docker, hence I have the following in `stack.yaml`.

```
docker:
  enabled: false
```

When trying to locate the produced product, using `stack --docker-image target/debian8.3 path --dist-dir`, I get the exactly same result as if I had omitted the `--docker-image` argument. To get the actual path for the products, I need to specify `--docker --docker-image target/debian8.3`.

So we have: 
```
stack --docker-image target/debian8.3 path --dist-dir == stack path --dist-dir
```
where I would have expected
```
stack --docker-image target/debian8.3 path --dist-dir == stack --docker --docker-image target/debian8.3 path --dist-dir
```

Should `--docker-image` not imply `--docker`?

**Comment by mgsloan** [comment]
Makes sense to me!

**Comment by angerman** [comment]
I'm not sure if this is part of a larger theme? Should maybe all `--docker-*` commands imply `--docker`?

**Comment by mgsloan** [comment]
Yeah, probably makes sense.  It gets a little weird with options like `--no-docker-registry-login`, though.

@borsboom What do you think about this?  Convenient, or too magical?

**Comment by borsboom** [comment]
I'm on the fence about this, but in a project's stack.yaml, the presence of a `docker:` section implies that docker is enabled, so it would be consistent if the presence of any `--docker*` argument did the same on the command-line.  Just make sure that if something like `--docker-image=foo --no-docker` is on the command-line, docker is still disabled (that the later `--no-docker` overrides the implied `--docker`).

**Comment by angerman** [comment]
My primary concern is that `--docker-image` without `--docker` seems to be a nonsensical construction. Maybe I'm seeing this wrong. But what does `--docker-image` without `--docker` or `--no-docker` actually mean?
